### ISSUE TITLE
a11y: Fix keyboard navigation and focus display issues on the combobox component

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#12795]
a11y Issues: 79, 20

**Issue 79**

Identical and inappropriate label is specified for the comboboxes resulted in screen reader users finding it difficult to understand the purpose of the comboboxes effectively.

Used in many places:

Preferences
Search for options label given for 
- Language  (LabeledSelect)
- Take me to cluster: (Select)
- Date format  (LabeledSelect)
- Time format  (LabeledSelect)

Create API Key
Search for options for 
- Scope (LabeledSelect)

Import YAML
Search for options for 
- Default Namespace (LabeledSelect)

Git Repo - Add Repository
Search for options label given for
- Watch (InputWithSelect - LabeledSelect)
- Git Authentication (SelectOrCreateAuthSecret - LabeledSelect)
- HELM authentication (SelectOrCreateAuthSecret - LabeledSelect)
-  TLS certificate verification (LabeledSelect)
- Target (which gets available on activating "Next" button) (LabeledSelect)

Pod Security Admission: Create
Search for options label given for
- Enforce (LabeledSelect  but missing label connection)
- Audit (LabeledSelect  but missing label connection)
- Warn (LabeledSelect  but missing label connection)

Repository: Create
Search for options label given for
- Authentication (which gets available on activating "http(s) URL to an index generated by Helm", "http(s) URL to an index generated by Helm" and "OCI Repository Experimental" radio buttons) (SelectOrCreateAuthSecret - LabeledSelect)

Import Extension Catalog
Search for options label given for
- Image Pull Secrets (LabeledSelect)

Banners
- Header Banner section
- Footer banner section
- Login screen Banner section
Search for options label given for
  - Font Size (LabeledSelect)
Example:

![image](https://github.com/user-attachments/assets/aad06664-d531-4901-8b53-a78e28fdcd14)

Remediation:

Apply the following changes:
• Remove "aria-label" attribute from source code of mentioned comboboxes.
• Specify "id" attribute with unique value to HTML elements containing text such as "Language", "Take me to cluster:", "Date Format", "Time Format" and "Table Rows per Page".
• Specify "aria-labelledby" attribute to <div > element containing combobox.
• Reference the "id" attribute values (of HTML elements) via "aria-labelledby" attribute respectively.

Issue 20 - Combobox displayed on keyboard focus

Combobox options gets displayed automatically when the mentioned combobox receives the keyboard focus resulting in keyboard-only and screen reader users will have to navigate through the options even when not required.

![image](https://github.com/user-attachments/assets/58bdd4e6-77b0-4889-be0e-af334403379e)

Remediiation:

Ensure that the combobox options do not get displayed automatically when the combobox receive the keyboard focus. It should get displayed when user activates the combobox.

**Comment by izaac** [comment]
Rancher version: `v2.11-0b081e077e830fa17b86151d21c786fb8d07c6de-head`
UI version: `master 0f09de2`
Firefox version: `135.0.1`
Chrome version: `133.0.6943.141`

Navigation over comboboxes doesn't expand automativally letting the user navigate to other areas without having to navigate through all the combobox options.
The options in the combobox are properly spelled out by a reader and the attributes requested in this issue's description are applied as expected.
